### PR TITLE
Add PDFgetN style normalization to SumSpectra

### LINKED ISSUE #21773: Add division option to sumspectra for bank merging
Add division by number of spectra as a normalisation option to SumSpectra when merging banks. The code for this can be found [here](https://github.com/neutrons/pdfgetn/blob/master/fortran/ublend4.f#L249).

This change should be an option (default to False) on the `SumSpectra` algorithm.

### PR DESCRIPTION
**Pull request by peterfpeterson**
This adds an option (default keeps old behavior) to multiply the output workspace by the number of spectra contributing. The reason for this option is that in PDFgetN, the data is normalized and multiplying by the number of contributing spectra does not preserve the correct behavior of the data.

*Report to:* @marshallmcdonnell

**To test:**

Run the following script and see the difference (most visible in low-Q) between `FkQ` and `FkQ_old`. Setting `MultiplyBySpectra=False` gives the appropriate summing for normalized data.
```python
import numpy as np

filename = 'Si_SRM_e_d91772.nxs'
# 27 appears to be S(Q) - 1
LoadNexusProcessed(Filename=filename, OutputWorkspace='banked', EntryNumber=27)

# reduce range
CropWorkspaceRagged(InputWorkspace='banked', OutputWorkspace='cropped',
                    Xmin=[0.67, 1.20, 2.42, 3.70, 4.12, 0.39],
                    Xmax=[10.20, 20.8, np.nan, np.nan, np.nan, 9.35])

# bin everything on a constant scale for summing
binning=(0.,.02,40.)
Rebin(InputWorkspace='cropped', OutputWorkspace='cropped',
          Params=binning)

# put into a single spectrum and Fourier transform
SumSpectra(InputWorkspace='cropped', OutputWorkspace='FkQ_old',
           WeightedSum=True, RemoveSpecialValues=True)
SumSpectra(InputWorkspace='cropped', OutputWorkspace='FkQ',
           WeightedSum=True, RemoveSpecialValues=True, MultiplyBySpectra=False)

PDFFourierTransform(InputWorkspace='FkQ', OutputWorkspace='Gr',
                    QMin=.67, # from above
                    InputSofQType='S(Q)-1', DeltaR=.02)
```

Fixes #21773.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by marshallmcdonnell — APPROVED** [review]
Testing of feature works as expected and described.

Also, PR includes code refactor and cleanup, proper documentation that explains the added features and when to use them appropriately (normalized vs. unnormalized data), and additional test coverage.

Overall, beneficial addition that is ready for merge.

File: Framework/Algorithms/src/SumSpectra.cpp [hunk 1]
```diff
@@ -146,6 +149,13 @@ void SumSpectra::init() {
                   "estimate the number of dropped values see the "
                   "description. ");
 
+  declareProperty("MultiplyBySpectra", true,
```
Comment (martyngigg) [inline Framework/Algorithms/src/SumSpectra.cpp]: Is this going to break the existing api if anyone has called `SumSpectra` with positional arguments in Python?

Comment (peterfpeterson) [reply Framework/Algorithms/src/SumSpectra.cpp]: I thought we had been telling people to use named arguments for a few years.

Comment (martyngigg) [reply Framework/Algorithms/src/SumSpectra.cpp]: We'd been recommending it but the api does still allow `*args`. As `SumSpectra` is a widely used algorithm I think we could conceivable have uses of this in scripts we don't know about.

**Pull request merged**